### PR TITLE
[top] Integrate adc_ctrl

### PR DESCRIPTION
**Pull request by tjaychen**
only last 3 commits are valid

### CODE REVIEW

File: hw/ip/dcd/data/dcd.hjson
Comment (tjaychen) [inline hw/ip/dcd/data/dcd.hjson:162]: @esshiu i'll file a separate issue for you to restore these and connect the unused signals.

**Review by alphan — APPROVED** [review]
Changes in dif_pwrmgr LGTM, thanks @tjaychen.

Quick question: Can I use the new source on FPGA after this change lands? Asking because it would unblock pwrmgr smoke tests.

**Comment by alphan** [comment]
@silvestrst can you take a quick look at the changes in PLIC DIF? Thanks!

**Comment by tjaychen** [comment]
> Changes in dif_pwrmgr LGTM, thanks @tjaychen.
> 
> Quick question: Can I use the new source on FPGA after this change lands? Asking because it would unblock pwrmgr smoke tests.

yes and no.... all the required components are hooked up, but pwrmgr and its associated wakeup components have not really been through much DV yet (it's on my list to write a smoke test). So you might run into **REALLY** stupid errors.  Do you want to file an issue against me to create said smoke test and go from there?

**Comment by alphan** [comment]
> > Changes in dif_pwrmgr LGTM, thanks @tjaychen.
> > Quick question: Can I use the new source on FPGA after this change lands? Asking because it would unblock pwrmgr smoke tests.
> 
> yes and no.... all the required components are hooked up, but pwrmgr and its associated wakeup components have not really been through much DV yet (it's on my list to write a smoke test). So you might run into **REALLY** stupid errors. Do you want to file an issue against me to create said smoke test and go from there?

Sounds good, thanks @tjaychen. Created #5592 and #5591.

File: hw/ip/dcd/data/dcd.hjson
Comment (msfschaffner) [inline hw/ip/dcd/data/dcd.hjson:144]: nit: there seem to be some non-ASCII apostrophes in this hjson, can you replace them with `'`?

**Review by msfschaffner — APPROVED** [review]
LGTM

**Comment by silvestrst** [comment]
> @silvestrst can you take a quick look at the changes in PLIC DIF? Thanks!

SW PLIC changes look good to me!

File: hw/ip/dcd/data/dcd.hjson
Comment (tjaychen) [reply hw/ip/dcd/data/dcd.hjson:144]: thanks for noticing that. removed.

**Pull request merged**
